[CI] MonotonicClockTests testMonotonicityWithSystemClock failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/6q3ezdpqppbts/tests/:x-pack:plugin:transform:test/org.elasticsearch.xpack.transform.transforms.scheduling.MonotonicClockTests/testMonotonicityWithSystemClock


**Reproduction line:**
```
./gradlew ':x-pack:plugin:transform:test' --tests "org.elasticsearch.xpack.transform.transforms.scheduling.MonotonicClockTests.testMonotonicityWithSystemClock" -Dtests.seed=FC9728F4DBFCFB30 -Dtests.locale=th-TH -Dtests.timezone=UTC -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.transform.transforms.scheduling.MonotonicClockTests#testMonotonicityWithSystemClock`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testMonotonicityWithSystemClock'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.transform.transforms.scheduling.MonotonicClockTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [elasticsearch[testHandleBulkResponseWithNoFailures][generic][T#1]][testHandleBulkResponseWithNoFailures]
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:194)
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:126)
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:122)
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:118)
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:114)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:994)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583), LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
#1:
	in [elasticsearch[testHandleBulkResponseWithNoFailures][generic][T#1]][testHandleBulkResponseWithNoFailures]
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:200)
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:133)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:1001)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583)
Created at:
	in [elasticsearch[testHandleBulkResponseWithNoFailures][generic][T#1]][testHandleBulkResponseWithNoFailures]
	org.elasticsearch.search.SearchHits.<init>(SearchHits.java:75)
	org.elasticsearch.search.SearchHits.<init>(SearchHits.java:57)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:994)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583), LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [elasticsearch[testHandleBulkResponseWithNoFailures][generic][T#1]][testHandleBulkResponseWithNoFailures]
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:88)
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:133)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:1001)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583), LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [elasticsearch[testHandleBulkResponseWithNoFailures][generic][T#1]][testHandleBulkResponseWithNoFailures]
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:194)
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:126)
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:122)
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:118)
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:114)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:994)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.onBulkResponse(AsyncTwoPhaseIndexer.java:581)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$onSearchResponse$11(AsyncTwoPhaseIndexer.java:547)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.ClientTransformIndexer.handleBulkResponse(ClientTransformIndexer.java:187)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextBulk(TransformIndexerFailureHandlingTests.java:217)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.onSearchResponse(AsyncTwoPhaseIndexer.java:533)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:200)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583), LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
#1:
	in [elasticsearch[testHandleBulkResponseWithNoFailures][generic][T#1]][testHandleBulkResponseWithNoFailures]
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:200)
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:133)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:1001)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.onBulkResponse(AsyncTwoPhaseIndexer.java:581)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$onSearchResponse$11(AsyncTwoPhaseIndexer.java:547)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.ClientTransformIndexer.handleBulkResponse(ClientTransformIndexer.java:187)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextBulk(TransformIndexerFailureHandlingTests.java:217)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.onSearchResponse(AsyncTwoPhaseIndexer.java:533)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:200)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583)
Created at:
	in [elasticsearch[testHandleBulkResponseWithNoFailures][generic][T#1]][testHandleBulkResponseWithNoFailures]
	org.elasticsearch.search.SearchHits.<init>(SearchHits.java:75)
	org.elasticsearch.search.SearchHits.<init>(SearchHits.java:57)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:994)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.onBulkResponse(AsyncTwoPhaseIndexer.java:581)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$onSearchResponse$11(AsyncTwoPhaseIndexer.java:547)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.ClientTransformIndexer.handleBulkResponse(ClientTransformIndexer.java:187)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextBulk(TransformIndexerFailureHandlingTests.java:217)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.onSearchResponse(AsyncTwoPhaseIndexer.java:533)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:200)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583), LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [elasticsearch[testHandleBulkResponseWithNoFailures][generic][T#1]][testHandleBulkResponseWithNoFailures]
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:88)
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:133)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:1001)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.onBulkResponse(AsyncTwoPhaseIndexer.java:581)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$onSearchResponse$11(AsyncTwoPhaseIndexer.java:547)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.ClientTransformIndexer.handleBulkResponse(ClientTransformIndexer.java:187)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextBulk(TransformIndexerFailureHandlingTests.java:217)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.onSearchResponse(AsyncTwoPhaseIndexer.java:533)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:200)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583), LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [elasticsearch[testHandleBulkResponseWithIrrecoverableFailures][generic][T#1]][testHandleBulkResponseWithIrrecoverableFailures]
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:194)
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:126)
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:122)
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:118)
	org.elasticsearch.search.SearchHit.<init>(SearchHit.java:114)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:994)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583), LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
#1:
	in [elasticsearch[testHandleBulkResponseWithIrrecoverableFailures][generic][T#1]][testHandleBulkResponseWithIrrecoverableFailures]
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:200)
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:133)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:1001)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583)
Created at:
	in [elasticsearch[testHandleBulkResponseWithIrrecoverableFailures][generic][T#1]][testHandleBulkResponseWithIrrecoverableFailures]
	org.elasticsearch.search.SearchHits.<init>(SearchHits.java:75)
	org.elasticsearch.search.SearchHits.<init>(SearchHits.java:57)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:994)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.common.AbstractCompositeAggFunction.getInitialProgressFromResponse(AbstractCompositeAggFunction.java:206)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$10(TransformIndexer.java:330)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.action.ActionListener.respondAndRelease(ActionListener.java:366)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetInitialProgress(TransformIndexerFailureHandlingTests.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$13(TransformIndexer.java:329)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.createCheckpoint(TransformIndexerFailureHandlingTests.java:186)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$14(TransformIndexer.java:311)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$16(TransformIndexer.java:373)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doGetFieldMappings(TransformIndexerFailureHandlingTests.java:287)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$18(TransformIndexer.java:380)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$21(TransformIndexer.java:418)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.onStart(TransformIndexer.java:450)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$5(AsyncTwoPhaseIndexer.java:230)
	org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:917)
	java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	java.base/java.lang.Thread.run(Thread.java:1583), LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [elasticsearch[testHandleBulkResponseWithIrrecoverableFailures][generic][T#1]][testHandleBulkResponseWithIrrecoverableFailures]
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:88)
	org.elasticsearch.action.search.SearchResponse.<init>(SearchResponse.java:133)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests.lambda$returnHit$26(TransformIndexerFailureHandlingTests.java:1001)
	org.elasticsearch.xpack.transform.transforms.TransformIndexerFailureHandlingTests$MockedTransformIndexer.doNextSearch(TransformIndexerFailureHandlingTests.java:199)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.triggerNextSearch(AsyncTwoPhaseIndexer.java:623)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.nextSearch(AsyncTwoPhaseIndexer.java:610)
	org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexer.lambda$maybeTriggerAsyncJob$4(AsyncTwoPhaseIndexer.java:233)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:248)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$6(TransformIndexer.java:301)
	org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
	org.elasticsearch.xpack.transform.transforms.TransformIndexer.lambda$onStart$7(TransformIndexer.java:333)
	org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:...
```

--- a/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerFailureHandlingTests.java
+++ b/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerFailureHandlingTests.java
@@ -992,7 +992,7 @@ public class TransformIndexerFailureHandlingTests extends ESTestCase {
 
     private static Function<SearchRequest, SearchResponse> returnHit() {
         return request -> new SearchResponse(
-            new SearchHits(new SearchHit[] { new SearchHit(1) }, new TotalHits(1L, TotalHits.Relation.EQUAL_TO), 1.0f),
+            SearchHits.unpooled(new SearchHit[] { SearchHit.unpooled(1) }, new TotalHits(1L, TotalHits.Relation.EQUAL_TO), 1.0f),
             // Simulate completely null aggs
             null,
             new Suggest(Collections.emptyList()),

--- a/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerStateTests.java
+++ b/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerStateTests.java
@@ -88,7 +88,7 @@ import static org.mockito.Mockito.verify;
 public class TransformIndexerStateTests extends ESTestCase {
 
     private static final SearchResponse ONE_HIT_SEARCH_RESPONSE = new SearchResponse(
-        new SearchHits(new SearchHit[] { new SearchHit(1) }, new TotalHits(1L, TotalHits.Relation.EQUAL_TO), 1.0f),
+        SearchHits.unpooled(new SearchHit[] { SearchHit.unpooled(1) }, new TotalHits(1L, TotalHits.Relation.EQUAL_TO), 1.0f),
         // Simulate completely null aggs
         null,
         new Suggest(Collections.emptyList()),

